mirage/unikernel.ml

```diff
@@ -47,6 +41,46 @@ module K = struct
         [ "dns-upstream" ]
     in
     Mirage_runtime.register_arg Arg.(value & opt (some string) None doc)
+
+  (* DNSmasq configuration options *)
+  (* TODO support multiple dhcp-range statements *)
+  let dhcp_range =
+    let doc = Arg.info ~doc:"Enable DHCP server." [ "dhcp-range" ] in
+    Mirage_runtime.register_arg
+      Arg.(value & opt Config_parser.(some dhcp_range_c) None doc)
+
+  let interface =
+    let doc = Arg.info ~doc:"Interface to listen on." [ "interface" ] in
+    Mirage_runtime.register_arg
+      Arg.(value & opt Config_parser.(some (ignore_c "interface")) None doc)
+
+  let except_interface =
+    let doc =
+      Arg.info ~doc:"Interface to not listen on." [ "except-interface" ]
+    in
+    Mirage_runtime.register_arg
+      Arg.(
+        value & opt Config_parser.(some (ignore_c "except-interface")) None doc)
+
+  let listen_address =
+    let doc = Arg.info ~doc:"IP address to listen on." [ "listen-address" ] in
+    Mirage_runtime.register_arg
+      Arg.(
+        value & opt Config_parser.(some (ignore_c "listen-address")) None doc)
+
+  let no_dhcp_interface =
+    let doc =
+      Arg.info ~doc:"Only provide DNS service on." [ "no-dhcp-interface" ]
+    in
+    Mirage_runtime.register_arg
+      Arg.(
+        value & opt Config_parser.(some (ignore_c "no-dhcp-interface")) None doc)
+
+  let bind_interfaces =
+    let doc =
+      Arg.info ~doc:"Bind to interface IP address only." [ "bind_interfaces" ]
+    in
+    Mirage_runtime.register_arg Arg.(value & flag doc)
```

Comment:
It could be nice if these didn't show up in the `--help` output since we don't do anything with them. I didn't find a way to achieve that.

Reply:
oh, we can introduce a specific section a la "dnsmasq options ignored by dnsvizor", and put these arguments there.